includes/class-edd-customer.php

```diff
@@ -334,12 +334,25 @@ public function update( $data = array() ) {
 
 		do_action( 'edd_customer_pre_update', $this->id, $data );
 
-		$updated = false;
+		$updated          = false;
+		$previous_user_id = $this->user_id;
 
 		if ( edd_update_customer( $this->id, $data ) ) {
 			$customer = edd_get_customer( $this->id );
 			$this->setup_customer( $customer );
 
+			// If the user ID changed, we need to update all the orders associated with this customer and change the user ID.
+			if ( intval( $previous_user_id ) !== intval( $this->user_id ) ) {
+
+				// Update some payment meta if we need to
+				$order_ids = edd_get_orders( array( 'customer_id' => $this->id, 'number' => 9999 ) );
```

Comment:
I assume this is a known fail-point for an edge case where a customer has more than 9999 payments. Is that accurate? (I'm not saying we need to do anything about that, just confirming)

Reply:
@mintplugins yeah, it's a known failure point, I'm 100% ok with it at this time.